test/framework/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregatorTestCase.java

```diff
@@ -185,23 +186,24 @@ private void testWithSeveralDocs(BooleanSupplier supplier, GeoBoundingBox bbox)
     }
 
     public void testSingletonDocsAsSubAgg() throws IOException {
-        testWithSeveralDocsAsSubAgg(() -> true, null);
+        testWithSeveralDocsAsSubAgg(() -> true, null, randomPrecision());
     }
 
     public void testBoundedSingletonDocsAsSubAgg() throws IOException {
-        testWithSeveralDocsAsSubAgg(() -> true, randomBBox());
+        int precision = randomPrecision();
+        testWithSeveralDocsAsSubAgg(() -> true, randomBBox(precision), precision);
     }
 
     public void testMultiValuedDocsAsSubAgg() throws IOException {
-        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, null);
+        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, null, randomPrecision());
     }
 
     public void testBoundedMultiValuedDocsAsSubAgg() throws IOException {
-        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, randomBBox());
+        int precision = randomPrecision();
+        testWithSeveralDocsAsSubAgg(LuceneTestCase::rarely, randomBBox(precision), randomPrecision());
```

Comment:
Why does this test generate two different random precisions, while `testNoundedSingletonDocsAsSubAgg` reuse the same precision twice?